platforms/core-execution/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonClient.java

```diff
@@ -80,23 +83,28 @@ public boolean isProcess(WorkerProcess workerProcess) {
     }
 
     public boolean isFailed() {
-        return failed;
+        return workerProcess.getExecResult().map(execResult -> execResult.getExitValue() > 0).orElse(false);
```

Comment:
I think we should use `!= 0` here, as I suspect Windows may return negative values in some cases (https://stackoverflow.com/a/74295941). Additionally, the Javadocs state that only `0` is a successful value by convention.

Reply:
Windows... 🤷 

Thanks for pointing this out.  Changed.